internal/controllers/storage/sync.go

```diff
@@ -354,5 +373,84 @@ func (r *Reconciler) getSecretKey(
 			secretKeyRef.Name,
 		)
 	}
+
 	return string(secretVal), nil
 }
+
+func (r *Reconciler) handlePauseResume(
+	ctx context.Context,
+	storage *resources.StorageClusterBuilder,
+) (bool, ctrl.Result, error) {
+	r.Log.Info("running step handlePauseResume for Storage")
+	if storage.Status.State == StorageReady && storage.Spec.Pause == PausedState {
+		r.Log.Info("`pause: Paused` was noticed, attempting to delete Storage StatefulSet")
+		statefulSet := &appsv1.StatefulSet{}
+		err := r.Client.Get(ctx,
+			types.NamespacedName{
+				Name:      storage.Name,
+				Namespace: storage.Namespace,
+			},
+			statefulSet,
+		)
+		if err != nil {
+			r.Log.Error(err, "failed to get the Storage StatefulSet object before deletion")
+			return Stop, ctrl.Result{RequeueAfter: DefaultRequeueDelay}, err
+		}
+
+		err = r.Client.Delete(ctx, statefulSet)
```

Comment:
Can we keep it simple and just change field `.spec.nodes`  to 0? We always use parallel [pod management policy](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#parallel-pod-management) in StatefulSet and as far i know do not have plans to change it. Deletion of StatefulSet is unnecessary destructive operation in this case